pkg/http-handler/http-handler.go

```diff
@@ -75,6 +77,35 @@ func getNamespaceAndCluster(req *http.Request) (string, string) {
 	return requestNamespace, requestCluster
 }
 
+// getHeaderNamespaces returns a list of namespaces from the header request
+// The name and the value of the header field is specified by 2 variables:
+// - headerName is a name of the expected header field, e.g. X-Consumer-Groups
+// - headerPattern is a regular expression and it matches only single regex group, e.g. ^namespace:([\w-]+)$
+func getHeaderNamespaces(req *http.Request, headerName, headerPattern string) ([]corev1.Namespace, error) {
+	var namespaces = []corev1.Namespace{}
+	if headerName == "" || headerPattern == "" {
+		return []corev1.Namespace{}, nil
+	}
+	r, err := regexp.Compile(headerPattern)
+	if err != nil {
+		log.Errorf("unable to compile regular expression: %v", err)
+		return namespaces, err
+	}
+	headerNamespacesOrigin := strings.Split(req.Header.Get(headerName), ",")
```

Comment:
@goock what happens if this returns an empty list?

Reply:
> @goock what happens if this returns an empty list?

nothing actually, when the slice is empty, the loop `for` just skips, and the entire function `getHeaderNamespaces` returns an empty list of the namespaces.